docs/vhost-user-blk-testing.md

```diff
@@ -0,0 +1,102 @@
+# How to test Vhost-user block with SPDK
+
+The purpose of this document is to illustrate how to test vhost-user-blk in cloud-hypervisor with SPDK as the backend.
+
+## Framework
+
+It's a simple test to validate the block read/write between VM and block backend.
+```
+             +----+----------+          +-------------+-----------+
+             |    |          |          |             |           |
+             |    |vhost-user|----------| vhost-user  |    dpdk   |
+             |    |blk device|          | port 1      |           |
+             |    |          |          |             |           |
+             |    +----------+          +-------------+-----------+
+             |               |          |                         |
+             |      vm       |          |           spdk          |
+             |               |          |                         |
+          +--+----------------------------------------------------+--+
+          |  |                  hugepages                         |  |
+          |  +----------------------------------------------------+  |
+          |                                                          |
+          |                       host                               |
+          |                                                          |
+          +----------------------------------------------------------+
+```
+## Prerequisites
+
+Prior to running the test, the following steps need to be performed.
+- Enable hugepages
+- Install SPDK
+
+Here are some good references for detailing them.
+- spdk
+	* https://spdk.io/doc/
+
+## Test environment
+
+The below test environment is based on ubuntu release, as for other system, please check related document.
+The test runs with multiple queue (MQ) support enabled, using 4 queues defined for both SPDK and the virtual machine. Here are the details on how the test can be run.
+
+# Host linux cmdline setting
+Add "default_hugepagesz=1G hugepagesz=1G hugepages=2" into /boot/grub/grub.cfg
+
+# Download the SPDK code
+git clone https://github.com/spdk/spdk
+cd spdk
+git submodule update --init
+
+# create the build dep
+./scripts/pkgdep.sh
+
+# build spdk
+./configure
+make
+
+# set the SPDk environment
```

Comment:
Strange capitalisation and inconsistent with above.

Reply:
Rob, thanks, i will change this.